base-template.sh

```diff
@@ -517,9 +517,15 @@ update_shared_hooks_if_appropriate() {
     GIT_NULL_REF="0000000000000000000000000000000000000000"
 
     RUN_UPDATE="false"
-    [ "$HOOK_NAME" = "post-merge" ] && RUN_UPDATE="true"
-    [ "$HOOK_NAME" = ".githooks.shared.trigger" ] && RUN_UPDATE="true"
-    [ "$HOOK_NAME" = "post-checkout" ] && [ "$1" = "$GIT_NULL_REF" ] && RUN_UPDATE="true"
+
+    if [ "$HOOK_NAME" = "post-merge" ] ||
+        { [ "$HOOK_NAME" = "post-checkout" ] && [ "$1" = "$GIT_NULL_REF" ]; } ||
+        {
+            git config --local githooks.sharedHooksUpdateTriggers
+            git config --global githooks.sharedHooksUpdateTriggers
+        } | grep "$HOOK_NAME"; then
```

Comment:
I'd prefer to avoid `{}` where we can.
You could change this from the existing code like
```
    [ "$HOOK_NAME" = "post-merge" ] && RUN_UPDATE="true"
    [ "$HOOK_NAME" = ".githooks.shared.trigger" ] && RUN_UPDATE="true"
    [ "$HOOK_NAME" = "post-checkout" ] && [ "$1" = "$GIT_NULL_REF" ] && RUN_UPDATE="true"
    git config --local githooks.sharedHooksUpdateTriggers | grep "$HOOK_NAME" && RUN_UPDATE="true"
    git config --global githooks.sharedHooksUpdateTriggers | grep "$HOOK_NAME" && RUN_UPDATE="true"
```

Reply:
What does {} actually do, it gathers output together IMO?  Is it a subshell? not necessaraly right? only in a pipe `| {... }` ?